src/rules/utils.ts

```diff
@@ -339,6 +339,7 @@ export enum ModifierName {
 export enum EqualityMatcher {
   toBe = 'toBe',
   toEqual = 'toEqual',
+  toMatchObject = 'toMatchObject',
```

Comment:
This isn't an equality matcher.

i.e `prefer-to-contain` will error on `expect(a.includes(b)).toMatchObject(true)`.

Instead just doing `matcher.name === 'toMatchObject'` should be enough.
(That'll actually improve things anyway, as the rule currently won't catch `toMatchObject` unless it's called & has only one argument).

Reply:
Removed in https://github.com/jest-community/eslint-plugin-jest/pull/545/commits/0f8ada9303b31b3e7936595529614921d90c64f4